cmd/dao_tests/pubkey.go

```diff
@@ -0,0 +1,208 @@
+	pkDao, err := dao.GetPubkeyDao(ctx)
+	if err != nil {
+		t.Errorf("Create pubkey test had failed.")
+		return
+	}
+	err = pkDao.Create(ctx, createPk()[0])
+	if err != nil {
+		t.Errorf("Create pubkey test had failed. %s", err)
+		return
+	}
+
+	pubkeys, err := pkDao.List(ctx, 10, 0)
+	if err != nil {
+		t.Errorf("Create pubkey test had failed. %s", err)
+		return
+	}
+
+	assert.Equal(t, 1, len(pubkeys), "Create pubkey test had failed.")
+}
+
+func ListPubkeyTest(t *testing.T) {
+	TearDown(t)
+	err := db.Seed("dev_small")
+	if err != nil {
+		t.Errorf("Error purging the database")
+		return
+	}
+
+	ctx := context.Background()
+	pkDao, err := dao.GetPubkeyDao(ctx)
+	if err != nil {
+		t.Errorf("List pubkey test had failed.")
+		return
+	}
+	pubkeys, err := pkDao.List(ctx, 100, 0)
+	if err != nil {
+		t.Errorf("List pubkey test had failed.")
+		return
+	}
+	assert.Equal(t, 1, len(pubkeys), "List Pubkey error.")
+}
+
+func UpdatePubkeyTest(t *testing.T) {
+	TearDown(t)
+	err := db.Seed("dev_small")
+	if err != nil {
+		t.Errorf("Error purging the database")
+		return
+	}
+
+	ctx := context.Background()
+	pkDao, err := dao.GetPubkeyDao(ctx)
+	if err != nil {
+		t.Errorf("Update pubkey test had failed.")
+		return
+	}
+
+	err = pkDao.Update(ctx, updatePk()[0])
+	if err != nil {
+		t.Errorf("Update pubkey test had failed. %s", err)
+		return
+	}
+
+	pubkeys, err := pkDao.List(ctx, 10, 0)
+	if err != nil {
+		t.Errorf("Update pubkey test had failed. %s", err)
+		return
+	}
+	assert.Equal(t, updatePk()[0].Name, pubkeys[0].Name, "Update pubkey test had failed.")
+}
+
+func GetPubkeyByIdTest(t *testing.T) {
+	TearDown(t)
+	err := db.Seed("dev_small")
+	if err != nil {
+		t.Errorf("Error purging the database")
+		return
+	}
+
+	ctx := context.Background()
+	pkDao, err := dao.GetPubkeyDao(ctx)
+	if err != nil {
+		t.Errorf("Get pubkey test had failed.")
+		return
+	}
+
+	pubkey, err := pkDao.GetById(ctx, 1)
+	if err != nil {
+		t.Errorf("Get pubkey test had failed.")
+		return
+	}
+	assert.Equal(t, "lzap-ed25519-2021", pubkey.Name, "Get Pubkey error.")
+	assert.Equal(t, "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIEhnn80ZywmjeBFFOGm+cm+5HUwm62qTVnjKlOdYFLHN lzap", pubkey.Body, "Get Pubkey error.")
+
+}
+
+func DeletePubkeyByIdTest(t *testing.T) {
+	TearDown(t)
+	err := db.Seed("dev_small")
+	if err != nil {
+		t.Errorf("Error purging the database")
+		return
+	}
+
+	ctx := context.Background()
+	pkDao, err := dao.GetPubkeyDao(ctx)
+	if err != nil {
+		t.Errorf("Delete pubkey test had failed.")
+		return
+	}
+
+	pubkeys, err := pkDao.List(ctx, 10, 0)
+	if err != nil {
+		t.Errorf("Delete pubkey test had failed")
+		return
+	}
+
+	err = pkDao.Delete(ctx, 1)
+	if err != nil {
+		t.Errorf("Delete pubkey test had failed")
+		return
+	}
+
+	pubkeysAfter, err := pkDao.List(ctx, 10, 0)
+	if err != nil {
+		t.Errorf("Delete pubkey test had failed")
+		return
+	}
+	assert.Equal(t, len(pubkeys)-1, len(pubkeysAfter), "Delete Pubkey error.")
+}
+
+func main() {
```

Comment:
Why haven't we used https://pkg.go.dev/testing@go1.18.4#hdr-Main I think then we would need to run `go test cmd/dao_tests` but not sure, did not test, but according to the docs, it is the function to be used for these cases, so I wonder why we have chosen to use internals of the testing library ([the docs for Main](https://pkg.go.dev/testing@go1.18.4#Main) say it stays exported as best effort, but no promises are made to keep it)